woocommerce/woocommerce-ios
# [IPPInAppFeedback] Banner business logic and merchant eligibility in Orders screen

Part of https://github.com/woocommerce/woocommerce-ios/issues/8530

## Description

This PR adds the logic to discern which in-app feedback banner we should show to the merchant, based on if the Store is eligible ( has COD enabled, and the country is IPP-supported ), and three conditions:

- Haven't collected any IPP payments.
- Have collected less than 10 IPP transactions in the last 30 days.
- Have collected more than 10 IPP transactions in the last 30 days.

At the moment we're not displaying or populating any specific content when any of the 3 conditions occur, this will be dealt with on a different PR.

## Changes
- Call to `displayIPPFeedbackBannerIfEligible()` in `OrderListViewController`'s init, which will perform the logic to check if COD is enabled, and fetch IPP transactions from storage (if any) as well as those within 30 days.

## Testing instructions
1. On a Store with IPP transactions, and COD enabled. Navigate to Menu > Orders
2. Check Xcode console and confirm that transactions within 30 days are logged:
```
hasResults? true
IPP transactions within 30 days: 3

[
  "OrderID: 1475 - PaymentMethodID: woocommerce_payments (WooCommerce In-Person Payments - DatePaid: Optional(2023-01-02 01:55:33 +0000)", 
  "OrderID: 1481 - PaymentMethodID: woocommerce_payments (WooCommerce In-Person Payments - DatePaid: Optional(2023-01-05 10:45:27 +0000)", 
  "OrderID: 1488 - PaymentMethodID: woocommerce_payments (WooCommerce In-Person Payments - DatePaid: Optional(2023-01-10 10:00:19 +0000)"
]

< 10 transactions within 30 days. Banner 2 shown
```

~~Due to the nature of fetching orders from Storage and how we use this layer as a sort of caching mechanism rather than as a source of truth, in some cases you may encounter 0 IPP transactions despite that these exist, for example if you install the app and go directly to the payments view without visiting the Orders view first. This is "expected" at this time, as we're looking for a fast solution to get us 90% there. The solution is to go to Orders > wait until are fetched > go back to Payments.~~

~~One potential way to smooth this problem is to add pull-to-refresh, this can be worked on a different PR or in i2 ( [tracked here](https://github.com/woocommerce/woocommerce-ios/issues/8530) )~~

Edit: As new requirements have changed the logic from Payments to Orders, this may no longer be a problem.
